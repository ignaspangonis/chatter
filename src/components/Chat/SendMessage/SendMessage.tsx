import React, { FormEvent, useState } from 'react'
import { Button, Form, InputGroup } from 'react-bootstrap'

type Props = {
  onSubmit: (message: string) => void
}

export default function SendMessage({ onSubmit }: Props) {
  const [message, setMessage] = useState('')

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault()

    if (!message) return

    onSubmit(message)
  }

  const handleMessageInput = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    setMessage(target.value)
  }

  return (
    <Form className="SendMessage" onSubmit={handleSubmit}>
      <InputGroup>
        <Form.Control placeholder="Aa" onChange={handleMessageInput} />

        <Button variant="primary" type="submit" disabled={!message}>
          Send
        </Button>
      </InputGroup>
    </Form>
  )
}
