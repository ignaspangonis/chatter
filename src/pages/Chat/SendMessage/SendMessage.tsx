import React, { FormEvent, useState } from 'react'

type Props = {
  onSubmit: (message: string) => void
}

export default function SendMessage({ onSubmit }: Props) {
  const [message, setMessage] = useState('')

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault()

    if (!message) return

    onSubmit(message)
    setMessage('')
  }

  const handleMessageInput = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    setMessage(target.value)
  }

  return (
    <form onSubmit={handleSubmit}>
      <div className="flex flex-row gap-regular grow">
        <input
          className="input input-bordered mx-auto flex-grow"
          value={message}
          placeholder="Aa"
          onChange={handleMessageInput}
        />

        <button className="btn" type="submit" disabled={!message}>
          Send
        </button>
      </div>
    </form>
  )
}
