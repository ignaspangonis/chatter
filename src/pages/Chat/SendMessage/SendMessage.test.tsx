import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { ComponentProps } from 'react'

import SendMessage from './SendMessage'

describe('<SendMessage {...props} />', () => {
  let props: ComponentProps<typeof SendMessage>

  beforeEach(() => {
    props = {
      onSubmit: jest.fn(),
    }
  })

  it('renders correctly', () => {
    const { container } = render(<SendMessage {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('displays user input', async () => {
    render(<SendMessage {...props} />)

    userEvent.type(screen.getByPlaceholderText('Aa'), 'Hello world!')

    expect(await screen.findByDisplayValue('Hello world!')).toBeInTheDocument()
  })

  it('calls callback on form submit', async () => {
    render(<SendMessage {...props} />)

    userEvent.type(screen.getByPlaceholderText('Aa'), 'Hello world!')
    userEvent.click(screen.getByText('Send'))

    await waitFor(() => expect(props.onSubmit).toHaveBeenCalledTimes(1))
    expect(props.onSubmit).toHaveBeenCalledWith('Hello world!')
  })
})
