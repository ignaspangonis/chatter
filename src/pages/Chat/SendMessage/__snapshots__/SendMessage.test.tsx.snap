// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendMessage {...props} /> renders correctly 1`] = `
<div>
  <form>
    <div
      class="flex flex-row gap-regular grow"
    >
      <input
        class="input input-bordered mx-auto flex-grow"
        placeholder="Aa"
        value=""
      />
      <button
        class="btn"
        disabled=""
        type="submit"
      >
        Send
      </button>
    </div>
  </form>
</div>
`;
